tests/unit/s2n_connection_serialize_test.c

```diff
@@ -465,8 +498,10 @@ int main(int argc, char **argv)
         EXPECT_OK(s2n_connections_set_io_stuffer_pair(new_client_conn, server_conn, &io_pair));
 
         /* Client can send and recv as usual */
-        EXPECT_OK(s2n_send_and_recv_test(server_conn, new_client_conn));
-        EXPECT_OK(s2n_send_and_recv_test(new_client_conn, server_conn));
+        for (size_t idx = 0; idx < 1000; idx++) {
+            EXPECT_OK(s2n_send_and_recv_test(server_conn, new_client_conn));
+            EXPECT_OK(s2n_send_and_recv_test(new_client_conn, server_conn));
+        }
```

Comment:
Do these tests fail without your fix? Wasn't the problem sending before the serialize, not after the deserialize?

Reply:
The problem is, if we send before serialization, then the failure will occur 3 encrypt calls after deserialization. These tests already send before serialization, I just had to make them do more than 2 encrypt calls after deserialization to get the failure case.